### PR TITLE
respect envSecretRefName coming from helm values

### PR DESCRIPTION
**Pull request by RafalSkolasinski**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Closes https://github.com/SeldonIO/seldon-core/issues/3640

**Special notes for your reviewer**:

In addition did some cleanup on Ansible in Core.

### CODE REVIEW

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **rafalskolasinski**
You can assign the PR to them by writing `/assign @rafalskolasinski` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by RafalSkolasinski** [comment]
All seems to work fine, I will look into adding some unit tests. 

However, is it okay to use secret from
```
predictiveUnit:
  defaultEnvSecretRefName: < secret here >
```
for explainers? Internally we refer to nodes of inference graph as predictive units. 

Personally, I think this is okay. I do not see need to introduce another field in helm values for explainer secrets.

**Comment by RafalSkolasinski** [comment]
integration tests passed 
image

**Comment by axsaucedo** [comment]
It seems to have quite a few ansible and prometheus changes, is this intended?

File: doc/source/analytics/analytics.md [hunk 1]
```diff
@@ -36,20 +36,20 @@ We recommend to configure Prometheus using [Prometheus Operator](https://github.
...
-kubectl create namespace seldon-system
+kubectl create namespace seldon-monitoring
 
-helm upgrade --install prometheus kube-prometheus \
-    --version 6.8.3 \
+helm upgrade --install seldon-monitoring kube-prometheus \
+    --version 6.9.5 \
 	--set fullnameOverride=seldon-monitoring \
-    --namespace seldon-system \
+    --namespace seldon-monitoring \
```
Comment (axsaucedo) [inline doc/source/analytics/analytics.md:44]: maybe we can just update to "prometheus" ns given it's no longer anything seldon specific

Comment (RafalSkolasinski) [reply doc/source/analytics/analytics.md:44]: Yes, the ansible in here needed a cleanup for quite some time. And on prometheus it is just a change of namespace in which example/docs installs it.

**Comment by RafalSkolasinski** [comment]
Yes, the ansible in here needed a cleanup for quite some time. And on prometheus it is just a change of namespace in which example/docs installs it.

**Comment by axsaucedo** [comment]
Ok - the pr looks good on the explainers envSecretRef, i added a comment on the prometheus, mainly on renaming the namespace

File: doc/source/analytics/analytics.md (hunk 1 above)
Comment (axsaucedo) [reply doc/source/analytics/analytics.md:44]: Currently we do suggest namespaces like `seldon-logs` so it would align for consistency so can go ahead

**Comment by axsaucedo** [comment]
Looks good
/approve

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[axsaucedo](https://github.com/SeldonIO/seldon-core/pull/4089#issuecomment-1123839026 "Approved")*

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)~~ [axsaucedo]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**